packages/fhir-converter/src/lib/outputProcessor/resourceDeduplicator.js

```diff
@@ -0,0 +1,98 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) 2022-present Metriport Inc.
+//
+// Licensed under AGPLv3. See LICENSE in the repo root for license information.
+//
+// This file incorporates work covered by the following copyright and
+// permission notice:
+//
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+//
+//     Permission to use, copy, modify, and/or distribute this software
+//     for any purpose with or without fee is hereby granted, provided
+//     that the above copyright notice and this permission notice appear
+//     in all copies.
+//
+//     THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+//     WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+//     WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+//     AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+//     CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+//     OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+//     NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+//     CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+// -------------------------------------------------------------------------------------------------
+
+var merge = require("./resourceMerger").merge;
+
+module.exports.Process = function (jsonObj) {
+  try {
+    const encounters = jsonObj.entry.filter(entry => {
+      return entry.resource.resourceType === "Encounter";
+    });
+
+    const idMap = new Set();
+    let duplicateId;
+    encounters.forEach(enc => {
+      enc.resource.identifier.forEach(id => {
+        if (idMap.has(id.value)) {
+          duplicateId = id.value;
+        } else {
+          idMap.add(id.value);
+        }
+      });
+    });
+
+    if (duplicateId) {
+      const duplicateEncounters = encounters.filter(enc => {
+        return enc.resource.identifier.some(id => id.value === duplicateId);
+      });
+      const enc1 = duplicateEncounters[0];
+      const enc2 = duplicateEncounters[1];
+      const mergedEncounter = merge(enc1, enc2);
+
+      const deduplicatedIdentifiers = deduplicateIdentifiers(mergedEncounter.resource.identifier);
+      mergedEncounter.resource.identifier = deduplicatedIdentifiers;
+      mergedEncounter.status = selectMostInformativeStatus(enc1.status, enc2.status);
+
+      const entriesWithoutEncounters = jsonObj.entry.filter(entry => {
```

Comment:
What's the entries without encounters logic? Can you explain more here just because I don't remember discussing this

Reply:
I should rename this variable to make it clearer. I'm just removing the two duplicated encounters here

Reply:
And then adding the one merged one instead